filetests/isa/x86/legalize-custom.clif

```diff
@@ -90,3 +90,56 @@ block0(v0: f32, v1: f32):
     ; check: $done(v2: f32):
     ; nextln: return v2
 }
+
+function %ineg_legalized_i8() {
+block0:
+    v0 = iconst.i8 1
+    v1 = ineg v0
+    ; check: v2 = iconst.i32 1
+    ; nextln: v0 = ireduce.i8 v2
+    ; nextln: v3 = iconst.i8 0
+    ; nextln: v4 = uextend.i32 v3
+    ; nextln: v5 = uextend.i32 v0
```

Comment:
Not sure I get why v2 and v3 have different types initially: shouldn't they both start out as `i8`s and be extended? And we have to extend and then reduce because we don't have an encoding for `isub.i8`? (@peterdelevoryas, no need to answer; this is more for @bjorn3 as he is more familiar with this than I am)

Reply:
iconst.i8 is legalized as iconst.i32 + ireduce.i8. isub.i8 is legalized as uextend.i32 + isub.i32 + ireduce.i8.